jobrunner/run.py

```diff
@@ -550,24 +576,43 @@ def set_code(job, code, message, error=None, results=None, timestamp_ns=None, **
             log.warning(
                 f"negative state duration of {duration}, clamping to 1ms\n"
                 f"before: {job.status_code:<24} at {ns_timestamp_to_datetime(job.status_code_updated_at)}\n"
-                f"after : {code:<24} at {ns_timestamp_to_datetime(timestamp_ns)}\n"
+                f"after : {new_status_code:<24} at {ns_timestamp_to_datetime(timestamp_ns)}\n"
             )
             timestamp_ns = int(job.status_code_updated_at + 1e6)  # set duration to 1ms
             timestamp_s = int(timestamp_ns // 1e9)
 
         # update coarse state and timings for user
-        if code in [StatusCode.PREPARED, StatusCode.PREPARING]:
+        if new_status_code in [StatusCode.PREPARED, StatusCode.PREPARING]:
             # we've started running
             job.state = State.RUNNING
             job.started_at = timestamp_s
-        elif code.is_final_code:
+        elif new_status_code in [StatusCode.CANCELLED_BY_USER]:
```

Comment:
I think the above change might remove the need for these if branches?

Them, we should only ever set CANCELLED_BY_USER code in one place, handling a FINALIZED job, in which case its just state.FAILED?

Reply:
agree, removed